src/interface/swarm.hpp

```diff
@@ -0,0 +1,150 @@
+                                      MeshBlock *pmb = nullptr);
+
+  /// Add variable to swarm
+  void Add(const std::string label, const Metadata &metadata);
+
+  /// Add multiple variables with common metadata to swarm
+  void Add(const std::vector<std::string> labelVector, const Metadata &metadata);
+
+  /// Remote a variable from swarm
+  void Remove(const std::string label);
+
+  /// Get real particle variable
+  ParticleVariable<Real> &GetReal(const std::string label) {
+    return *(realMap_.at(label));
+  }
+
+  /// Get integer particle variable
+  ParticleVariable<int> &GetInteger(const std::string label) {
+    return *(intMap_.at(label));
+  }
+
+  /// Assign label for swarm
+  void setLabel(const std::string label) { label_ = label; }
+
+  /// retrieve label for swarm
+  std::string label() const { return label_; }
+
+  /// retrieve metadata for swarm
+  const Metadata metadata() const { return m_; }
+
+  /// Assign info for swarm
+  void setInfo(const std::string info) { info_ = info; }
+
+  /// return information string
+  std::string info() { return info_; }
+
+  /// Expand pool size geometrically as necessary
+  void increasePoolMax() { setPoolMax(2 * nmax_pool_); }
+
+  /// Set max pool size
+  void setPoolMax(const int nmax_pool);
+
+  /// Check whether metadata bit is set
+  bool IsSet(const MetadataFlag bit) const { return m_.IsSet(bit); }
+
+  /// Get the last index of active particles
+  int get_max_active_index() { return max_active_index_; }
+
+  /// Get number of active particles
+  int get_num_active() { return num_active_; }
+
+  /// Get the quality of the data layout. 1 is perfectly organized, < 1
+  /// indicates gaps in the list.
+  Real get_packing_efficiency() { return num_active_ / (max_active_index_ + 1); }
+
+  bool mpiStatus;
+
+  /// Mark particle for removal (usually during a parallel dispatch)
+  KOKKOS_INLINE_FUNCTION
+  void MarkParticleForRemoval(int index) { marked_for_removal_(index) = true; }
+
+  /// Remove particles marked for removal and update internal indexing
+  void RemoveMarkedParticles();
+
+  /// Open up memory for new empty particles, return a mask to these particles
+  ParArrayND<bool> AddEmptyParticles(int num_to_add);
+
+  /// Defragment the list by moving active particles so they are contiguous in
+  /// memory
+  void Defrag();
+
+ private:
+  int nmax_pool_;
+  int max_active_index_ = 0;
+  int num_active_ = 0;
+  Metadata m_;
+  std::string label_;
+  std::string info_;
+  std::shared_ptr<ParArrayND<PARTICLE_STATUS>> pstatus_;
+  ParticleVariableVector<int> intVector_;
+  ParticleVariableVector<Real> realVector_;
+
+  MapToParticle<int> intMap_;
+  MapToParticle<Real> realMap_;
+
+  std::list<int> free_indices_;
+  ParticleVariable<bool> mask_;
+  ParticleVariable<bool> marked_for_removal_;
+};
+
+using SP_Swarm = std::shared_ptr<Swarm>;
+using SwarmVector = std::vector<SP_Swarm>;
+using SwarmMap = std::map<std::string, SP_Swarm>;
```

Comment:
Same comment about map vs unordered_map. I just want to know the justification?

Reply:
Made this change so `Swarm` can serve as an inspiration to the continuum parts of parthenon :) 

(I agree, often when I use a `map`, `unordered_map` seems to be preferred, although how it hashes entries and what guarantees on efficiency there are is still unclear to me).